invenio_drafts_resources/services/records/service.py

```diff
@@ -263,6 +266,21 @@ def update_draft(
         """Replace a draft."""
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
 
+        # can not make record restricted after grace period
+        if (
+            not current_app.config["ALLOW_RECORD_RESTRICTION_AFTER_GRACE_PERIOD"]
+            and data.get("access", {}).get("record", None) == "restricted"
+            and draft.get("access", {}).get("record", None) == "public"
+        ):
+            end_of_grace_period = (
+                draft.created + current_app.config["RECORDS_RESTRICTION_GRACE_PERIOD"]
+            )
```

Comment:
for better conditional statement treadability:
```suggestion
        # can not make record restricted after grace period
        current_draft_is_public = draft.get("access", {}).get("record", None) == "public"
        is_update_to_restricted = data.get("access", {}).get("record", None) == "restricted"
        allow_restriction =  current_app.config["ALLOW_RECORD_RESTRICTION_AFTER_GRACE_PERIOD"]
        
        if not allow_restriction and current_draft_is_public and is_update_to_restricted:
            end_of_grace_period = (
                draft.created + current_app.config["RECORDS_RESTRICTION_GRACE_PERIOD"]
            )
```